### ISSUE TITLE
[YouTube] Error snackbar when watching some videos (NPE in YoutubeStreamExtractor.getStreamSegments())

### ISSUE THREAD

**Issue by qube-03** [TeamNewPipe/NewPipe#7792]
### Checklist

- [x] I am using the latest version - 0.21.16 
- [ x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [ ] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [ x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug

1. Go to literally any video and play it
2. watch the video without problem but with the popup (inconsistent across videos)

### Actual behavior

i can watch videos just fine, but at the bottom it always tells me  "Sorry, something went wrong" in a popup

### Expected behavior

watch the video without the error at the bottom

## Exception
* __User Action:__ requested stream
* __Request:__ https://www.youtube.com/watch?v=xhVS1HKwGWw
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ YouTube
* __Version:__ 0.21.16
* __OS:__ Linux samsung/r8qxeea/r8q:11/RP1A.200720.012/G781BXXU4CUH5:user/release-keys 11 - 30
<details><summary><b>Crash log </b></summary><p>

```
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.getStreamSegments(YoutubeStreamExtractor.java:1335)
	at org.schabi.newpipe.extractor.stream.StreamInfo.extractOptionalData(StreamInfo.java:333)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:73)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:64)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getStreamInfo$3(ExtractorHelper.java:116)
	at org.schabi.newpipe.util.ExtractorHelper.$r8$lambda$YTHJjScxCJNO1LTCqs3IKy35iyY(Unknown Source:0)
	at org.schabi.newpipe.util.ExtractorHelper$$ExternalSyntheticLambda6.call(Unknown Source:4)
	at io.reactivex.rxjava3.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:43)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.rxjava3.core.Maybe.subscribe(Maybe.java:5330)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:15753)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.rxjava3.core.Maybe.subscribe(Maybe.java:5330)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:614)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)

```
</details>
<hr>

### Device info

 - Android version/Custom ROM version:
 - Device model:

**Comment by qube-03** [comment]
this also hsppens when i fullscreen/unfullscreen videos

**Comment by TheMCNerd2017** [comment]
I've been encountering this same issue as well for a few days now (even before the YouTube change that broke viewing videos). Seems to be completely random when the error appears too. Not sure if it's related to the changes to the n-sig challenge Google has done a few days ago.

**Comment by TiA4f8R** [comment]
The issue is caused by this new `engagementPanelSectionListRenderer` element returned sometimes by YouTube on the `next` endpoint of the InnerTube API:


```json
{
  "engagementPanelSectionListRenderer": {
    "content": {
      "adsEngagementPanelContentRenderer": {
        "hack": true
      }
    },
    "targetId": "engagement-panel-ads",
    "visibility": "ENGAGEMENT_PANEL_VISIBILITY_HIDDEN",
    "loggingDirectives": {
      "trackingParams": "removed",
      "visibility": {
        "types": "12"
      }
    }
  }
}
```

Contrary to all other `engagementPanelSectionListRenderer`seen, it doesn't have a `panelIdentifier` (contrary to the one of chapters (`engagement-panel-macro-markers-description-chapters`) and transcript (`engagement-panel-transcript`) panels).

The extractor right now uses the panel identifier name to check if its the chapters panel or not, so a `NullPointerException` is thrown [here](https://github.com/TeamNewPipe/NewPipeExtractor/blob/65129e65a81563c81cd8dbcb6c64e7570bd2910c/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java#L1275-L1276):

```java
                if (panelIdentifier.equals(
                        "engagement-panel-macro-markers-description-chapters")) {
```

This issue can be easily fixed by adding a null check on `panelIdentifier`.

**Comment by qube-03** [comment]
i dont have the cosing skills to do thst myself so as of right now ill keep this open until a bugfix release

**Comment by seniorm0ment** [comment]
encountering this myself, it's just insanely annoying, at minimim the ability to just hide the notification because everything is working fine for me.

**Comment by mhmdanas** [comment]
@seniorm0ment if you're not aware, this was fixed in NewPipe 0.22.0.
